content/blog/js-beginners-guide.md

```diff
@@ -0,0 +1,107 @@
+---
+author: "Alex Hill"
+date: 2020-02-02
+title: Front-end Javascript Frameworks - A Beginner's Guide
+tags:
+ - javascript
+ - vue
+ - react
+ - angular
+---
+
+We've put together a little resource for RSEs (or anyone else!) who are writing browser 
+based apps and want to be using the latest Javascript web technologies, but aren't sure 
+where to start. 
+
+The full <a href="/resources/jsbeginnersguide.pdf" target="_blank">poster</a> was presented 
+at [RSLondonSouthEast](https://rslondon.ac.uk/) on 6th February.
+
+## Key concepts 
+If you've just gotten started with the world of front-end frameworks, you've probably heard of 
+**components** and **directives**. These are the key building blocks for modern JS frameworks.
+
+A component is like a custom HTML element with special behaviour attached:
+
+```
+<error-list errors="myErrorArray"></error-list>
+```
+
+A directive is like a custom data attribute on a regular HTML element, with special behaviour attached:
+
+```
+<span v-if="hasError" v-text="message"></span>
+```
+
+A component is a combination of a **template** that defines the presentation, and a script that defines 
+behaviour.
+
+Here's an example of an HTML template in Vue.js. It uses the v-for directive to loop over an array:
+
+```
+<ul class="list">
+    <li v-for="e in errors">
+        {{e}}
+    </li>
+</ul>
+```
+
+And here's how you'd achieve the same thing in React. React doesn't support HTML templates but instead 
+uses **JSX**, a syntax extension to Javascript that resembles HTML:
+
+```
+<ul className="list">
+    {errors.map(e => <li>{e}</li>)}
+</ul>
+```
+
+## Application architecture
+Finally, you might have heard of [Flux](https://facebook.github.io/flux/) - it's a code architecture pattern pioneered by Facebook, the creators of React,
+ and now widely adopted for organising complex front-end applications. React by itself is just a lightweight library for writing components, but it is commonly 
+used in conjunction with [Redux](https://react-redux.js.org/) which is a Flux implementation. 
+
+The best way to understand Flux is probably to read the [docs](https://facebook.github.io/flux/docs/overview) over at Facebook. 
+But in a nutshell, it's a uni-directional data flow pattern inspired by functional programming.
```

Comment:
Can you expand the nutshell a bit? "Uni-directional" seems to be the important word here but I don't understand why that is important in this context